### ISSUE TITLE
The user_find command does not respect all=True or raw=True

### ISSUE THREAD

**Issue by Aethylred** [waldur/python-freeipa#1]
It you set up a freeipa-python Client object to communicate with a directory server and use the `all=True` and/or the `raw=True` options it provides the same output regardless if they're set to true or false or omitted.

Given configured client object:

```
len(client.user_find()['result'][0])
```

and 

```
len(client.user_find(all=True)['result'][0])
```

These two codes give identical outputs with 20 attributes (I'm not pasting the raw output with personal information keys and codes).

Using the ipa console:

```
(Custom IPA interactive Python console)
>>> len(api.Command.user_find()['result'][0])
12
>>> len(api.Command.user_find(all=True)['result'][0])
23
>>>
```

Clearly freeipa-python is not provided the expected extra attributes.

**Comment by Aethylred** [comment]
I'm seeing similar issues with `user_show` where the I expect the `all=True` flag to reveal the `hasPassword` attribute.

**Comment by AmbientLighter** [comment]
Hi @Aethylred!

> Given configured client object:
> 
> len(client.user_find()['result'][0])
> and
> 
> len(client.user_find(all=True)['result'][0])
> These two codes give identical outputs with 20 attributes (I'm not pasting the raw output with personal information keys and codes).

Yes, we're passing all=True by default in both user_find and user_show: https://github.com/opennode/python-freeipa/blob/develop/src/python_freeipa/client.py#L176

However, you may override default value by passing all=False in user_find, for example:

`len(client.user_find(all=False)['result'][0])`

> Clearly freeipa-python is not provided the expected extra attributes.

Could you please check if it makes difference to pass raw=False in user_find, for example:

`len(client.user_find(all=False, raw=False)['result'][0])`

**Comment by Aethylred** [comment]
Hello @AmbientLighter 

We found the issue, the user calling user_find has to have permission to read the attributes. Adding the appropriate attributes to **System: Read User Kerberos Login Attributes** makes them visible.

My issue was that admin was getting the desired attributes, but accounts with the **User Administrator** role, which lacked the required permissions, was not able to see them.

Now that I've fixed the permissions python-freeipa is working as expected.
